src/Algebra/Graph/Class.hs

```diff
@@ -170,6 +171,48 @@ instance Ord a => Graph (R.Relation a) where
     overlay = R.overlay
     connect = R.connect
 
+-- TODO: Optimise the implementation by caching the results of symmetric closure.
+instance Ord a => Graph (RID.ReflexiveRelation a) where
```

Comment:
Why are you moving around `ReflexiveRelation` and some other instances which are unrelated to this PR? This may be the right thing to do, but it should be discussed in a separate PR.

Reply:
I moved it because I needed to move the `SymmetricRelation` instance and since I had the trouble to change one I thought I should change the others but my bad and I'm sorry for doing it! Do you want me to open a PR related to this change?

Reply:
Don't be sorry -- it may be the right thing to do, but I think it may be better to do it in a separate PR, after we finish with this one :) 

Why did you need to move the `SymmetricRelation` instance?

Reply:
Because when I was implementing the tests for the `SymmetricRelation` I import the `SymmetricRelation` and `InternalDerived` module (to use the basic primitives and such) and it would create a dependency loop with the `Class` module if I remember correctly. So in order to fix that and be able to use the basic primitives I needed to get rid of that loop and moved all the instances to the `Class` file

Reply:
I don't understand how testing imports can lead to cycles. You can import anything from the testsuite, because the library code does not import anything from the testsuite.

I think you can probably leave all instances where they are currently and there should be no cycles.

Also, note, that it may be better to define `SymmetricRelation` as a separate data type, without wrapping a `newtype` around `Relation`, because you actually don't seem to reuse any code from `Relation`, right? This will allow you avoid all unnecessary wrapping and unwrapping.

Reply:
If I wanted to make use of the basic primitives and `(*)`/`(+)` I had to import the `Class` module on the `InternalDerived` module in order to implement the `Num` and `Graph` instances; if I'm not mistaken that's how it is written (in your version). That caused a import loop.. I'm sorry but I can't remember what files caused it but the culprit was the `InternalDerived` because I needed to remove the `import Algebra.Graph.Class` from it.

I do reuse a lot of the Relation code to implement 90% of the functions.

Reply:
> I do reuse a lot of the Relation code to implement 90% of the functions.

Aha, interesting. In that case let's keep the `newtype`.

Reply:
Looks like you still haven't undone changes to unrelated instances like `ReflexiveRelation`.